docs/HowToDebug.md

```diff
@@ -1,4 +1,80 @@
+
+## Different kinds of errors
+
+There are different kinds of errors. However, they can be divided into three categories based on their severity. So when nni fails, check each part sequentially.
+
+Generally, if webUI is started successfully, there is a `Status` in the `Overview` tab, serving as a possible indicator of what kind of error happens. Otherwise you should check manually.
+
+### **NNI** Fails
+
+This is the most serious error. When this happens, the whole experiment fails and no trial will be run. Usually this might be related to some installation problem.
+
+When this happens, you should check `nnictl`'s error output file `stderr` and then the `nnimanager`'s log to find if there is any error.
+
+
+### **Dispatcher** Fails
+
+Dispatcher fails. Usually, for some new users of NNI, it means that tuner fails. You could check dispatcher's log to see what happens to your dispatcher. For built-in tuner, some common errors might be invalid search space (unsupported type of search space or inconsistence between initializing args in configuration file and actual tuner's \_\_init\_\_ function args).
+
+Take the later situation as an example. If you write a customized tuner who's \_\_init\_\_ function has an argument called `optimize_mode`, which you do not provide in your configuration file, NNI will fail to run your tuner so the experiment fails. You can see errors in the webUI like:
+
+![](img/dispatcher_error.jpg)
+
+Here we can see it is a dispatcher error. So we can check dispatcher's log, which might look like:
+
+```
+[2019-02-19 19:36:45] DEBUG (nni.main/MainThread) START
+[2019-02-19 19:36:47] ERROR (nni.main/MainThread) __init__() missing 1 required positional arguments: 'optimize_mode'
+Traceback (most recent call last):
+  File "/usr/lib/python3.7/site-packages/nni/__main__.py", line 202, in <module>
+    main()
+  File "/usr/lib/python3.7/site-packages/nni/__main__.py", line 164, in main
+    args.tuner_args)
+  File "/usr/lib/python3.7/site-packages/nni/__main__.py", line 81, in create_customized_class_instance
+    instance = class_constructor(**class_args)
+TypeError: __init__() missing 1 required positional arguments: 'optimize_mode'.
+```
+
+### **Trial** Fails
+
+In this situation, NNI can still run and dispatch trials. 
+
+It means your trial code (which is run by NNI) fails. This kind of error is strongly related to your trial code. Please check trial's log to fix any possible errors shown there.
+
+A common example of this would be run the mnist example without installing tensorflow. Surely there is an Import Error (that is, not installing tensorflow but trying to import it in your trial code) and thus every trial fails.
+
+![](img/trial_error.jpg)
+
+As it shows, every trial has a log path, where you can find trial'log and stderr.

```

Comment:
could you also show how to configure log level here, so that it is easy for users to debug.

Reply:
Have added in `Where it the log`